pr 25.02.2023

danya1705 opened the pull request.

danya1705 committed:
написаны Апдейт-сервис и Статус бота

danya1705 committed:
написаны Апдейт-сервис и Статус бота

Barsev174 committed:
Добавил классы: TelegramBotUpdatesListener, NotificationTaskRepository, NotificationTask, TelegramBotConfiguration. Пока без логики

Barsev174 committed:
Добавил классы: TelegramBotUpdatesListener, NotificationTaskRepository, NotificationTask, TelegramBotConfiguration. Пока без логики

danya1705 committed:
Merge branch 'sergei-feature' into dev

# Conflicts:
#	pom.xml

IGuruI-1981 committed:
Внесены данные БЫ в app.properties
Добавлен файл chengelog и user (liquibase)

IGuruI-1981 committed:
Внесены данные БЫ в app.properties
Добавлен файл chengelog и user (liquibase)

IGuruI-1981 committed:
Внесены данные БЫ в app.properties
Добавлен файл chengelog и user (liquibase)

IGuruI-1981 committed:
Внесены данные БЫ в app.properties
Добавлен файл chengelog и user (liquibase)

IGuruI-1981 committed:
Внесены изменения в таблицу, файл user.sql

IGuruI-1981 committed:
Внесены изменения в таблицу, файл user.sql

danya1705 committed:
маленькие доработки

danya1705 committed:
маленькие доработки

IGuruI-1981 committed:
Добавлены зависимости для работы с БД и liquibase

IGuruI-1981 committed:
Добавлены зависимости для работы с БД и liquibase

IGuruI-1981 committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	pom.xml

danya1705 committed:
маленькие доработки

danya1705 committed:
маленькие доработки

Modusen committed:
Добавлены классы для быстрого поключения клавиатур на 4 кнопки под нужную ситуацию в двух вариациях

Modusen committed:
Добавлены классы для быстрого поключения клавиатур на 4 кнопки под нужную ситуацию в двух вариациях

IGuruI-1981 committed:
Добавлена сущность UserCustodian и репозиторий UserCustodianRepository

IGuruI-1981 committed:
Добавлена сущность UserCustodian и репозиторий UserCustodianRepository

IGuruI-1981 committed:
Merge branch 'viktor-feature' into dev

IGuruI-1981 committed:
Внесены измененния в UserCustodian и  UserCustodianRepository

IGuruI-1981 committed:
Внесены измененния в UserCustodian и  UserCustodianRepository

IGuruI-1981 committed:
Удалены дубликаты зависимостей

IGuruI-1981 committed:
Удалены дубликаты зависимостей

dmitry-bizin commented on pom.xml:
```diff
@@ -22,11 +22,41 @@
+
+        <dependency>
+            <groupId>com.github.pengrad</groupId>
+            <artifactId>java-telegram-bot-api</artifactId>
```
Зачем второй раз подключаете? выше уже подключили же

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/service/UpdateService.java:
```diff
@@ -0,0 +1,66 @@
+    /**
+     * Ключ - идентификатор Телеграм-чата. Значение - статус общение клиент-бот для данного чата.
+     */
+    public Map<Long, BotStatus> statusMap = new HashMap<>();
```
private

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/service/UpdateService.java:
```diff
@@ -0,0 +1,66 @@
+    public SendMessage updateHandler (Update update) {
+
+        Long chatId = update.message().chat().id();
+        statusMap.putIfAbsent(chatId, BotStatus.PRINT_GREETINGS_MESSAGE);
```
не очень понял смысл в этом. Зачем всегда при любом сообщении класть по ключу сообщение PRINT_GREETINGS_MESSAGE, чтобы потом на следующей строке его получить?

IGuruI-1981 committed:
Добавлена сущность Animal

danya1705 committed:
Добавлен класс с контроллерами

danya1705 committed:
Добавлен класс с контроллерами

danya1705 committed:
Добавлен класс с контроллерами

IGuruI-1981 committed:
Добавлена сущность Report
Добавлены таблицы Animal&Report

IGuruI-1981 committed:
Добавлены репозитории AnimalRepository & ReportRepository

IGuruI-1981 committed:
Внесены изменения в таблицу Report

IGuruI-1981 committed:
Merge branch 'viktor-feature' into dev

danya1705 committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	pom.xml
#	src/main/resources/liquibase/sql/user.sql

Modusen committed:
Изменены версии зависимостей для работы Swagger.
Добалвен API на взаимодействие с испытательным сроком. Функционал рабочий.
Изменены импорты Виктора для работы приложения.
Поправлена таблица в User.sql у Виктора, добавлена своя новая.

danya1705 committed:
Merge branch 'dmitriy-feature' into dev

# Conflicts:
#	pom.xml
#	src/main/java/pro/sky/java/course7/animal_shelter_bot/controller/AnimalShelterController.java

danya1705 committed:
Сделаны контроллеры и сервис по работе с отчетами и напоминаниями. Дополнена база report.

Barsev174 committed:
Добавил контроллер для зактрытия пробного периода

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/entity/TrialPeriod.java:
```diff
@@ -0,0 +1,57 @@
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private Date StartDate;
```
Названия полей в camelCase

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/entity/TrialPeriod.java:
```diff
@@ -0,0 +1,57 @@
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private Date StartDate;
+    private Date EndDate;
```
Названия полей в camelCase

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/service/ReportService.java:
```diff
@@ -0,0 +1,42 @@
+            reportList.addAll(reportRepository.findReportsByParentId(id));
+        }
+        return reportList.stream()
+                .filter(e -> e.getReportDate().isAfter(dateFrom) && e.getReportDate().isBefore(dateTo))
```
Такое лучше фильтровать на уровне запроса к БД, а не в коде

IGuruI-1981 committed:
Внесены изменения в БД, в модели

Barsev174 committed:
Немного поправил эндпоинт по удалению

Barsev174 committed:
Немного поправил эндпоинт по удалению

danya1705 committed:
Написан тест для выдачи отчетов волонтёру.
Сделан рефакторинг UpdateService и KeyboardService, добавлен класс Buttons, теперь в боте доступен переход в меню Этапа 1 и Этапа 2 и выход из них.

danya1705 committed:
Merge branch 'daniil-feature' into dev

danya1705 committed:
Адаптация кода под измененные поля в БД

danya1705 committed:
Добавил тест для updateTrialPeriod

danya1705 committed:
AnimalShelterController разделён на TrialPeriodController и ReportController

Barsev174 committed:
Добавил контроллер для Custodian'a(добавление, редактирование, вывод всех)

Barsev174 committed:
Добавлены комментарии в CustodianController

IGuruI-1981 committed:
Добавлены контроллеры и севисы в AnimalController & AnimalService. Добавлено исключение AnimalNotFoundException

IGuruI-1981 committed:
Merge branch 'viktor-feature' into dev

IGuruI-1981 committed:
Создан пакет exception, исключение AnimalNotFoundException перенесено в пакет exception

IGuruI-1981 committed:
Дописаны API к AnimalController

IGuruI-1981 committed:
Добавлена таблица volunteer

IGuruI-1981 committed:
Добавлены модель Volunteer, VolunteerController, VolunteerRepositiory, VolunteerService

Barsev174 committed:
Добавил Exсeption в контроллеры

IGuruI-1981 committed:
Внесены изменения в таблицу trial_period
Внесены изменения в ReportController, ReportService, TrialPeriod, TrialPeriodRepository

IGuruI-1981 committed:
Merge branch 'viktor-feature' into dev

danya1705 committed:
Поправлены ReportService и ReportController после изменений в БД
Восстановлена работа кнопок
Восстановлена работа тестов

IGuruI-1981 committed:
Update VolunteerController.java

Обновление данных в VolunteerController

IGuruI-1981 committed:
Update VolunteerService.java

Внесение изменений в VolunteerService

IGuruI-1981 committed:
Create VolunteerNotFoundException

Создан VolunteerNotFoundException

IGuruI-1981 committed:
Rename VolunteerNotFoundException to VolunteerNotFoundException.java

Внесены изменения в VolunteerNotFoundException

IGuruI-1981 committed:
Update VolunteerNotFoundException.java

Внесены изменения

dmitry-bizin commented on pom.xml:
```diff
@@ -22,11 +22,48 @@
+
+        <dependency>
+            <groupId>io.springfox</groupId>
+            <artifactId>springfox-swagger-ui</artifactId>
```
Эта библиотека - другая реализация OpenAPI, у вас сейчас springdoc и springfox - выберите одну, потому что они обе про одно и то же

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/controller/AnimalController.java:
```diff
@@ -0,0 +1,77 @@
+                    content = {@Content(mediaType = "application/json",
+                            schema = @Schema(implementation = Animal.class))})
+    })
+    @PostMapping("/add")
```
никаких слов писать не нужно, т.к. сам метод определяет действие

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/controller/AnimalController.java:
```diff
@@ -0,0 +1,77 @@
+                    content = {@Content(mediaType = "application/json",
+                            schema = @Schema(implementation = Animal.class))})
+    })
+    @DeleteMapping("delete/{id}")
```
никаких слов писать не нужно, т.к. сам метод определяет действие

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/controller/AnimalController.java:
```diff
@@ -0,0 +1,77 @@
+                    content = {@Content(mediaType = "application/json",
+                            schema = @Schema(implementation = Animal.class))})
+    })
+    @PutMapping("/update")
```
никаких слов писать не нужно, т.к. сам метод определяет действие

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/controller/CustodianController.java:
```diff
@@ -0,0 +1,70 @@
+                    content = {@Content(mediaType = "application/json",
+                            schema = @Schema(implementation = UserCustodian.class))})
+    })
+    @GetMapping("/getAllCustodian")
```
никаких слов писать не нужно, т.к. сам метод определяет действие

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/controller/CustodianController.java:
```diff
@@ -0,0 +1,70 @@
+                    content = {@Content(mediaType = "application/json",
+                            schema = @Schema(implementation = UserCustodian.class))})
+    })
+    @PutMapping("/update")
```
никаких слов писать не нужно, т.к. сам метод определяет действие

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/controller/CustodianController.java:
```diff
@@ -0,0 +1,70 @@
+                    content = {@Content(mediaType = "application/json",
+                            schema = @Schema(implementation = UserCustodian.class))})
+    })
+    @PostMapping("/newCustodian")
```
никаких слов писать не нужно, т.к. сам метод определяет действие

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/controller/TrialPeriodController.java:
```diff
@@ -0,0 +1,83 @@
+                    content = {@Content(mediaType = "application/json",
+                            schema = @Schema(implementation = TrialPeriod.class))})
+    })
+    @GetMapping("/getAll")
```
никаких слов писать не нужно, т.к. сам метод определяет действие

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/controller/TrialPeriodController.java:
```diff
@@ -0,0 +1,83 @@
+                    content = {@Content(mediaType = "application/json",
+                            schema = @Schema(implementation = TrialPeriod.class))})
+    })
+    @PostMapping("/new")
```
никаких слов писать не нужно, т.к. сам метод определяет действие

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/controller/TrialPeriodController.java:
```diff
@@ -0,0 +1,83 @@
+                    content = {@Content(mediaType = "application/json",
+                            schema = @Schema(implementation = TrialPeriod.class))})
+    })
+    @PutMapping("/new")
```
никаких слов писать не нужно, т.к. сам метод определяет действие

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/model/TrialPeriod.java:
```diff
@@ -0,0 +1,68 @@
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private LocalDate StartDate;
```
названия полей должны начинаться с маленькой буквы

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/model/TrialPeriod.java:
```diff
@@ -0,0 +1,68 @@
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private LocalDate StartDate;
+    private LocalDate EndDate;
```
названия полей должны начинаться с маленькой буквы

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/service/AnimalService.java:
```diff
@@ -0,0 +1,46 @@
+
+    public Animal deleteAnimalId(Long id) {
+        Animal animalDelete = animalRepository.findById(id).orElse(null);
+        if (animalDelete != null) {
```
можно вместо orElse сделать через ifPresent в функциональном стиле

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/service/CustodianService.java:
```diff
@@ -0,0 +1,35 @@
+
+    public UserCustodian editCustodian(UserCustodian custodian) {
+        UserCustodian findCustodian = userCustodianRepository.findById(custodian.getId()).orElse(null);
+        if (findCustodian != null) {
```
можно вместо orElse сделать через ifPresent в функциональном стиле

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/service/TrialPeriodService.java:
```diff
@@ -0,0 +1,40 @@
+
+@Service
+public class TrialPeriodService {
+    TrialPeriodRepository trialPeriodRepository;
```
private final

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/service/TrialPeriodService.java:
```diff
@@ -0,0 +1,40 @@
+
+    public TrialPeriod editTrialPeriod(TrialPeriod period) {
+        TrialPeriod findTrialPeriod = trialPeriodRepository.findById(period.getId()).orElse(null);
+        if (findTrialPeriod == null) {
```
можно вместо orElse сделать через ifPresent в функциональном стиле

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/service/TrialPeriodService.java:
```diff
@@ -0,0 +1,40 @@
+
+    public TrialPeriod deletePeriod(long id) {
+        TrialPeriod findTrialPeriod = trialPeriodRepository.findById(id).orElse(null);
+        if (findTrialPeriod != null) {
```
можно вместо orElse сделать через ifPresent в функциональном стиле

IGuruI-1981 committed:
Внесены изменения в VolunteerController и VolunteerService

IGuruI-1981 committed:
Merge branch 'viktor-feature' into dev

Modusen committed:
Добавлен функционал проверки первого контакта Юзера с ботом.
Добавлена кнопка запроса контакта, если контакт с Юзером первый
Добавлена регистрация пользователя в БД при передаче контакта по кнопке.

Modusen committed:
Merge branch 'dev' into dmitriy-feature

Modusen committed:
Косметические изменения

danya1705 committed:
Покрыты тестами TelegramBotUpdatesListener и UpdateService

Barsev174 committed:
Добавил выбор питомца

Barsev174 committed:
Merge remote-tracking branch 'origin/dev' into dev

Barsev174 committed:
Добавил информацию для кнопок

Barsev174 committed:
Merge branch 'sergei-feature' into dev

danya1705 committed:
Полностью покрыты тестами все контроллеры за исключением последних эндпойнтов в каждом из них.
Добавлены equals и hashCode в сущности.

Modusen committed:
Функционал отправки отчетов и внесения отчетов в базы данных.

danya1705 committed:
Изменил инъекции в ReportService, убрал ReportCRUDService

danya1705 committed:
Добавлен класс Photo и реализован вывод фотографий из отчета в Swagger.
Исправлена функция выдачи списка отчетов.

Barsev174 committed:
Реализовал работу кнопок в меню информации о животных

Barsev174 committed:
Merge branch 'dev' into sergei-feature

Modusen committed:
Реализована проверка на существование пользователя, что не позволит создать в бд несколько одинаковых пользователей после отправки контакта.

danya1705 committed:
Доработаны тесты

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/model/Animal.java:
```diff
@@ -0,0 +1,66 @@
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    private String animalType;
```
тип животного лучше сделать enum-ом

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/model/ReplyButton.java:
```diff
@@ -0,0 +1,15 @@
+public enum ReplyButton {
+    SEND_CONTACT_BUTTON("Отправить контакт");
+
+    String text;
```
private final

Barsev174 committed:
Добавил логику на три кнопки в информации о приюте

Barsev174 committed:
Убрал кнопки "оставить контакт"

IGuruI-1981 committed:
Внесены изменения в VolunteerController, VolunteerService, VolunteerRepository и др.

IGuruI-1981 committed:
Merge branch 'dev' into viktor-feature

# Conflicts:
#	src/main/java/pro/sky/java/course7/animal_shelter_bot/model/BotStatus.java
#	src/main/java/pro/sky/java/course7/animal_shelter_bot/service/UpdateService.java

danya1705 committed:
Добавлен тест выдачи фотографий в контроллере отчетов

danya1705 committed:
Добито покрытие тестами UpdateService до 90%

Modusen committed:
сохранение отчетов сделано с помощью мапы во избежание конфликтов при одновременном заполнении отчетов разными юзерами.

danya1705 committed:
Закончено покрытие тестами UpdateService и ReportController

IGuruI-1981 committed:
Merge branch 'dev' into viktor-feature

IGuruI-1981 committed:
Добавлены меню позвать волонтера

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/controller/VolunteerController.java:
```diff
@@ -0,0 +1,89 @@
+                    content = {@Content(mediaType = "application/json",
+                            schema = @Schema(implementation = Volunteer.class))})
+    })
+    @PostMapping("/add")
```
никаких слов писать не нужно, т.к. сам метод определяет действие

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/controller/VolunteerController.java:
```diff
@@ -0,0 +1,89 @@
+                    content = {@Content(mediaType = "application/json",
+                            schema = @Schema(implementation = Volunteer.class))})
+    })
+    @DeleteMapping("delete/{id}")
```
никаких слов писать не нужно, т.к. сам метод определяет действие

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/controller/VolunteerController.java:
```diff
@@ -0,0 +1,89 @@
+                    content = {@Content(mediaType = "application/json",
+                            schema = @Schema(implementation = Volunteer.class))})
+    })
+    @PutMapping("/update")
```
никаких слов писать не нужно, т.к. сам метод определяет действие

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/controller/VolunteerController.java:
```diff
@@ -0,0 +1,89 @@
+                    content = {@Content(mediaType = "application/json",
+                            schema = @Schema(implementation = Volunteer.class))})
+    })
+    @GetMapping("/all")
```
никаких слов писать не нужно, т.к. сам метод определяет действие

dmitry-bizin commented on src/main/java/pro/sky/java/course7/animal_shelter_bot/service/ReportService.java:
```diff
@@ -0,0 +1,102 @@
+        if (chatId.isPresent()) {
+            SendMessage message = new SendMessage(chatId.get(), text);
+            SendResponse response = telegramBot.execute(message);
+            System.out.println("Sending message to " + chatId);
```
logger

danya1705 committed:
Доработан стиль кода по замечаниям в пулл-реквесте

Barsev174 committed:
Добавил комментарии US

danya1705 committed:
Дописаны тесты последних эндпойнтов в контроллерах

Modusen committed:
Добавлены оповещения волонтеров о завершающихся испытательных сроках по шедулеру.
Добавлены эндпоинты по продлению испытательного срока. Также добавлена рассылка сообщений при закрытии(удалении) или продлении испытательного срока.

danya1705 committed:
Дописаны тесты для новых эндпойнтов из TrialPeriodController

IGuruI-1981 committed:
Добавлена реализация меню поиск волонтера

IGuruI-1981 committed:
Merge branch 'viktor-feature' into dev

# Conflicts:
#	src/main/java/pro/sky/java/course7/animal_shelter_bot/service/UpdateService.java

danya1705 committed:
Дописаны тесты для UpdateService

danya1705 committed:
Добавлена проверка на получение нулевого объекта Photo от Телеграма
